HandyHansel/Formatters/CategoryHelpFormatter.cs

```diff
@@ -0,0 +1,135 @@
+                            .Append("`")
+                            .Append(ca.Name)
+                            .Append(" (")
+                            .Append(this.CommandsNext.GetUserFriendlyTypeName(ca.Type))
+                            .Append(")` - ")
+                            .Append(opt ? "optional - " : "")
+                            .Append(ca.Description ?? "No description provided. Please contact the dev team.")
+                            .Append("\n");
+                    }
+                    allUsageStrings.Append("`\n");
+                }
+                this._embed.AddField("Usage:", allUsageStrings.ToString(), false);
+                if (command.Overloads.Any(co => co.Arguments.Any()))
+                {
+                    this._embed.AddField("Arguments:", allArguments.ToString(), false);
+                }
+            }
+
+            // TODO: Add the permissions needed to use this command. 
+
+            return this;
+        }
+
+        public override BaseHelpFormatter WithSubcommands(IEnumerable<Command> subcommands)
+        {
+            Dictionary<string, ISet<string>> modules = new Dictionary<string, ISet<string>>();
+            List<CommandGroup> groupCommands = new List<CommandGroup>();
+            List<Command> currentLevelCommands = new List<Command>();
+
+            foreach (Command command in subcommands.OrderBy(test => test.Name))
+            {
+                if (command.CustomAttributes.FirstOrDefault(item => item is BotCategoryAttribute) is BotCategoryAttribute module)
+                {
+                    if (!modules.ContainsKey(module.Name))
+                    {
+                        modules[module.Name] = new HashSet<string>();
+                    }
+                    modules[module.Name].Add(command.Name);
+                }
+                else if (command is CommandGroup cgroup)
+                {
+                    groupCommands.Add(cgroup);
+                }
+                else
+                {
+                    currentLevelCommands.Add(command);
+                }
+            }
+
+            if (currentLevelCommands.Any())
+            {
+                this._embed.AddField(this._command == null ? "Commands:" : "Sub-commands:", string.Join(", ", currentLevelCommands.Select(x => Formatter.InlineCode(x.Name))), true);
+            }
+
+            if (groupCommands.Any())
+            {
+                this._embed.AddField(this._command == null ? "Groups:" : "Sub-groups:", string.Join(", ", groupCommands.Select(x => Formatter.InlineCode(x.Name))), true);
+            }
+
+            if (modules.Any())
+            {
+                foreach (KeyValuePair<string, ISet<string>> moduleLists in modules)
+                {
+                    this._embed.AddField(moduleLists.Key, string.Join(", ", moduleLists.Value.Select(Formatter.InlineCode)), true);
+                }
+            }
+
+            return this;
+        }
+    }
+}
```

Comment:
Define special category help formatter that allows you to split commands into categories for easier visual understanding of the kinds of commands available.